src/audio-worklet/migration/spn-recorder/app.js

```diff
@@ -90,8 +101,9 @@ function setupScriptProcessor(recordBuffer, passSampleToVisualizers) {
 
       // While recording, feed data to recording buffer at the proper time.
       if (isRecording) {
-        recordBuffer
-            .copyToChannel(currentSamples[channel], channel, recordingLength);
+        let frameNumber = recordingLength / BUFFER_SIZE;
```

Comment:
With this operation, `frameNumber` can be a float. Is that okay?

Reply:
JS don't have the different number type like int, float, or double. The only type is number. 

Based on my design, the recordingLength is increased by BUFFER_SIZE each time (line 115). Therefore, recordingLength/BUFFER_SIZE will always be an "INTEGER"

Reply:
> JS don't have the different number type like int, float, or double. The only type is number.

Yes. I am well aware of that. (and that can be a "bad part" for JS audio programming) The reason why I am asking this is that when this `frameNumber` becomes a floating point number it creates a confusion when we use this to pinpoint a sample value in the array. 

```js
a = [1, 2, 3]
a[0.75] // undefined
```

The resulting value of `undefined` will create `NaN` values in the audio stream for subsequent operation. (That's very bad.) So it is a general practice in audio programming to make it clear that the sample index number is a positive integer.

> Based on my design, the recordingLength is increased by BUFFER_SIZE each time (line 115). Therefore, recordingLength/BUFFER_SIZE will always be an "INTEGER"

You have two options:
1) Add a comment that explains why this operation is safe for sample indexing.
2) Use `Math.floor()` to make sure it falls on to an integer.

Does it make sense?